src/Types.ts

```diff
@@ -49,7 +49,7 @@ export interface IndexDefinition {
 }
 
 export interface Document {
```

Comment:
JSDoc valuable here

Reply:
Perhaps add a DataTransferObject interface and relate it to this. Reference in `MongoRepository` type params and `Repository.spec.ts` extensions like `UserDTO`. 

Et voila! Documentation through interfaces.